File: 2023/java/src/test/java/org/frieder/aoc/day17/TestDay17A.java
package org.frieder.aoc.day17;

import org.frieder.aoc.day17.a.Solution17A;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertEquals;

public class TestDay17A {

    @Test
    public void testExample() throws IOException {
        assertEquals(102, Solution17A.getResult("../resources/17/test_a.txt"), 0);
    }

    @Test
    public void testResult() throws IOException {
        assertEquals(1263, Solution17A.getResult("../resources/17/input.txt"), 0);
    }
}

File: 2023/php/test/TestDay17A.php
<?php declare(strict_types=1);

$loader = require __DIR__ . '/../vendor/autoload.php';

use day17\a\Solution17A;
use PHPUnit\Framework\TestCase;
use utils\TestLogger;

final class TestDay17A extends TestCase
{

    public function testExample(): void
    {
        $this->assertSame(102, Solution17A::getResult("../resources/17/test_a.txt", TestLogger::createTestLogger()));
    }

    public function testResult(): void
    {
        $this->assertSame(1263, Solution17A::getResult("../resources/17/input.txt", TestLogger::createTestLogger()));
    }

}